streams/src/main/java/org/apache/kafka/streams/processor/internals/namedtopology/KafkaStreamsNamedTopologyWrapper.java

```diff
@@ -16,27 +16,84 @@
  */
 package org.apache.kafka.streams.processor.internals.namedtopology;
 
+import org.apache.kafka.common.annotation.InterfaceStability.Evolving;
 import org.apache.kafka.streams.KafkaClientSupplier;
 import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.errors.TopologyException;
+import org.apache.kafka.streams.processor.internals.TopologyMetadata;
 
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.Properties;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
 
+/**
+ * This is currently an internal and experimental feature for enabling certain kinds of topology upgrades. Use at
+ * your own risk.
+ *
+ * Status: basic architecture implemented but no actual upgrades are supported yet
+ *
+ * Note: some standard features of Kafka Streams are not yet supported with NamedTopologies. These include:
+ *       - global state stores
+ *       - interactive queries (IQ)
+ *       - TopologyTestDriver (TTD)
```

Comment:
Just want to call this out in case you come across one of these things and wonder if it will work with Named Topologies -- it won't 🙂